tests/p2p/test_sync_rate_limiter.py

```diff
@@ -0,0 +1,54 @@
+from unittest.mock import MagicMock
+
+from hathor.simulator import FakeConnection
+from hathor.simulator.trigger import StopAfterNMinedBlocks
+from tests import unittest
+from tests.simulation.base import SimulatorTestCase
+
+
+class BaseRandomSimulatorTestCase(SimulatorTestCase):
+    def test_sync_rate_limiter(self):
+        manager1 = self.create_peer()
+
+        miner1 = self.simulator.create_miner(manager1, hashpower=10e6)
+        miner1.start()
+        trigger = StopAfterNMinedBlocks(miner1, quantity=20)
+        self.simulator.run(3600, trigger=trigger)
+
+        manager2 = self.create_peer()
+        manager2.connections.MAX_ENABLED_SYNC = 0
+        conn12 = FakeConnection(manager1, manager2, latency=0.05)
+        self.simulator.add_connection(conn12)
+        self.simulator.run(3600)
+
+        manager2.connections.enable_rate_limiter()
+
+        connected_peers2 = list(manager2.connections.connected_peers.values())
+        self.assertEqual(1, len(connected_peers2))
+        protocol2 = connected_peers2[0]
+        sync2 = protocol2.state.sync_manager
+        sync2._send_tips = MagicMock()
+
+        for _ in range(8):
+            sync2.send_tips()
+            self.assertTrue(sync2._send_tips.called)
+            sync2._send_tips.reset_mock()
+
+        sync2.send_tips()
+        self.assertFalse(sync2._send_tips.called)
+
+        self.simulator._clock.advance(2000)
+        self.assertTrue(sync2._send_tips.called)
```

Comment:
It's better to assert `call_count` instead of `called`, so you don't get false successes. By changing the range to `100`, we can see that `send_tips` is actually called 100 times, catching a bug:

```suggestion
        for _ in range(100):
            sync2.send_tips()

        self.assertEqual(sync2._send_tips.call_count, 8)

        sync2.send_tips()
        self.assertEqual(sync2._send_tips.call_count, 8)

        self.simulator._clock.advance(2000)
        self.assertEqual(sync2._send_tips.call_count, 9)
```

Reply:
Much better. Done! Thanks!